website/identifiers/utils.py

```diff
@@ -125,10 +130,8 @@ def parse_identifiers(ezid_response):
             'ark': '{0}{1}'.format(settings.ARK_NAMESPACE.replace('ark:', ''), suffix),
         }
     else:
-        return dict(
-            [each.strip('/') for each in pair.strip().split(':')]
-            for pair in resp['success'].split('|')
-        )
+        return {'doi': resp['success'].strip().split(':')[1]}
```

Comment:
Instead of relying on an index for pulling out the DOI (I'm just nervous about the service ever changing the order of their response) what about leaving the first response as a dict and explicitly taking the `doi` key:

```
identifiers = dict([each.strip('/') for each in pair.strip().split(':')] for pair in resp['success'].split('|'))
return identifiers['doi']
```

Reply:
It should return {'doi': '.......'}, but for identifiers['doi'], the return value is a str instead of a dict. Rebuild a dict like return {'doi': identifiers['doi']} is not worthy.